src/wms/src/main/java/org/geoserver/wms/kvp/FilterKvpParser.java

```diff
@@ -75,12 +76,28 @@ public Object parse(String value) throws Exception {
                 Filter filter;
                 final byte[] rawContent = string.getBytes();
                 InputStream input = new ByteArrayInputStream(rawContent);
-
+                if (string.contains(FES.NAMESPACE)) {
+                    try {
+                        input = new ByteArrayInputStream(rawContent);
+                        Configuration configuration =
+                                new org.geotools.filter.v2_0.FESConfiguration();
+                        Parser parser_2_0_0 = new Parser(configuration);
+                        parser_2_0_0.setEntityResolver(resolverProvider.getEntityResolver());
+                        parser_2_0_0.setValidating(true);
+                        filter = (Filter) parser_2_0_0.parse(input);
+                        filters.add(filter);
+                    } catch (Exception e3) {
+                        String msg = "Unable to parse filter: " + string;
+                        LOGGER.log(Level.WARNING, msg, e3);
```

Comment:
I would not recommend logging this as a warning, but rather at debug level. WFS 1.1.0 requests are still quite common, this means that for each WFS 1.1.0 request, we would have a warning in the log, which will not be a warning.

I would suggest that you struct the passing method for WFS per version, something like this.

```java

private static final class ParsingResult {
  Filter filter;
  Exception exception. 
}

private ParsingResult parseWFS_2_0_0(String rawFilter) {
  ParsingResult result = ....   
  try {
   ...
  result.filter = filter;
  } catch (Exception exception) {
   result.exception = exception.
  }
 return result;
}
```

Then at the end fo the main method, if no parsing was successful (you will have a parse result for each version of WFS), you can log all the exceptions you got at the warning level saying that the parameter could not be parsed. Otherwise, you just returned the parsed filter.